src/main/g8/src/main/scala/$package$/UserRoutes.scala

```diff
@@ -1,90 +1,74 @@
 package $package$
 
-import akka.actor.{ ActorRef, ActorSystem }
-import akka.event.Logging
-
 import scala.concurrent.duration._
 import akka.http.scaladsl.server.Directives._
 import akka.http.scaladsl.model.StatusCodes
 import akka.http.scaladsl.server.Route
-import akka.http.scaladsl.server.directives.MethodDirectives.delete
-import akka.http.scaladsl.server.directives.MethodDirectives.get
-import akka.http.scaladsl.server.directives.MethodDirectives.post
-import akka.http.scaladsl.server.directives.RouteDirectives.complete
-import akka.http.scaladsl.server.directives.PathDirectives.path
 
 import scala.concurrent.Future
-import $package$.UserRegistryActor._
-import akka.pattern.ask
+import $package$.UserRegistry._
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.scaladsl.AskPattern._
 import akka.util.Timeout
 
+//#import-json-formats
 //#user-routes-class
-trait UserRoutes extends JsonSupport {
+class UserRoutes(userRegistry: ActorRef[UserRegistry.Command])(implicit val system: ActorSystem[_]) {
   //#user-routes-class
+  import JsonFormats._
+  //#import-json-formats
 
-  // we leave these abstract, since they will be provided by the App
-  implicit def system: ActorSystem
-
-  lazy val log = Logging(system, classOf[UserRoutes])
-
-  // other dependencies that UserRoutes use
-  def userRegistryActor: ActorRef
-
-  // Required by the `ask` (?) method below
-  implicit lazy val timeout = Timeout(5.seconds) // usually we'd obtain the timeout from the system's configuration
+  // Required by the `ask` method below
+  // in a real application we'd likely want to obtain the timeout from the app configuration
+  // with something like `system.settings.config.getDuration("my-app.routes.ask-timeout")`
+  private implicit val timeout = Timeout(5.seconds)
 
   //#all-routes
   //#users-get-post
-  //#users-get-delete   
-  lazy val userRoutes: Route =
+  //#users-get-delete
+  val userRoutes: Route =
     pathPrefix("users") {
       concat(
         //#users-get-delete
         pathEnd {
           concat(
             get {
-              val users: Future[Users] =
-                (userRegistryActor ? GetUsers).mapTo[Users]
+              val users: Future[Users] = userRegistry.ask(GetUsers)
```

Comment:
Maybe make this a method `def getUsers(): Future[Users]` below (or even in the companion of UserRegistry). Then, `complete(getUsers())` might work here well enough.

Reply:
I think having methods here makes most sense as this is the piece of code that wants the abstraction. I'll add such methods (if we decide to keep the actor that is)